lib/reply.js

```diff
@@ -158,6 +155,9 @@ function wrapPumpCallback (reply) {
 
 function wrapReplyEnd (reply, payload) {
   reply.sent = true
+  if (!reply.res.getHeader('Content-Length')) {
+    reply.res.setHeader('Content-Length', Buffer.byteLength(payload))
+  }
```

Comment:
you need to move back this piece, or add an existential check for `payload`. https://github.com/fastify/fastify/pull/246/files#diff-d2fdf63c1d7f1d9f68270b6dfa8c930cL154 calls this function without it.

Reply:
Nice catch!
I'm wondering why it fails only in Node ≥ 7 🤔